### PR TITLE
Fix odd even

### PR DESCRIPTION
**Pull request by sauloperez**
#### What? Why?

This fixes https://github.com/openfoodfoundation/openfoodnetwork/issues/1710 by removing unnecessary duplication of #destroy. We can make it work by adding a [simple fix in Spree](https://github.com/coopdevs/spree/pull/2).

To do that though it refactors things a bit. It extracts the JSON responses from the admin's line item controller, which are only used by the bulk management feature, into its own controller. This decouples spree from an OFN-only feature and allows to remove unnecessary code. Furthermore, `Admin::LineItemsController` is gone in Spree 2.4.0.beta. See: spree/spree#5280. So the sooner we get rid of it, the better.

#### What should we test?

The alternate backgrounds plus bulk order management

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -10,7 +10,7 @@ gem 'i18n-js', '~> 3.0.0'
 gem 'nokogiri', '>= 1.6.7.1'
 
 gem 'pg'
-gem 'spree', github: 'coopdevs/spree', branch: 'spree-upgrade-step-6'
+gem 'spree', github: 'coopdevs/spree', branch: 'fix-odd-even'
```
Comment (sauloperez) [inline Gemfile]: update me when that branch gets merged into `spree-upgrade-step-6`.

File: spec/controllers/admin/bulk_line_items_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+          end
+
+          it 'returns a 204 response' do
+            spree_put :update, params
+            expect(response.status).to eq 204
+          end
+
+          it 'applies enterprise fees locking the order with an exclusive row lock' do
+            expect_any_instance_of(Spree::Order).to receive(:reload).with(lock: true)
+            expect_any_instance_of(Spree::Order).to receive(:update_distribution_charge!)
```
Comment (sauloperez) [inline spec/controllers/admin/bulk_line_items_controller_spec.rb:182]: I can't understand why but I can't make it pass with `expect(order1).to ...` although the instance is indeed the same that receives the method calls. Any idea? I'm sure I'm missing something.

Comment (enricostano) [reply spec/controllers/admin/bulk_line_items_controller_spec.rb:182]: I don't like this 😭

**Review by enricostano — CHANGES_REQUESTED** [review]

File: spec/controllers/admin/bulk_line_items_controller_spec.rb (hunk 2 above)
Comment (enricostano) [reply spec/controllers/admin/bulk_line_items_controller_spec.rb:182]: Opened an issue to fix this https://github.com/openfoodfoundation/openfoodnetwork/issues/1762

**Review by enricostano — APPROVED** [review]

**Pull request merged**